tests/suites/constraints/constraints.sh

```diff
@@ -1,3 +1,32 @@
+run_constraints_lxd() {
+  # Echo out to ensure nice output to the test suite.
+  echo
+
+  file="${TEST_DIR}/constraints-lxd.txt"
+
+  ensure "constraints-lxd" "${file}"
+
+  juju add-machine --constraints "cores=2"
+  juju add-machine --constraints "cores=2 mem=2G"
+
+  wait_for_machine_agent_status "0" "started"
+  wait_for_machine_agent_status "1" "started"
+
+	echo "Ensure machine 0 has 2 cores"
+  machine0_hardware=$(juju machines --format json | jq -r '.["machines"]["0"]["hardware"]')
+  machine0_cores=$(echo "$machine0_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /cores/){print $i}}}')
+  check_ge "${machine0_cores}" "cores=2"
+
+	echo "Ensure machine 1 has 2 cores and 2G memory"
+  machine1_hardware=$(juju machines --format json | jq -r '.["machines"]["1"]["hardware"]')
+  machine1_cores=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /cores/){print $i}}}')
+  machine1_mem=$(echo "$machine1_hardware" | awk '{for(i=1;i<=NF;i++){if($i ~ /mem/){print $i}}}')
+  check_ge "${machine1_cores}" "cores=2"
+  check_ge "${machine1_mem}" "mem=2048M"
```

Comment:
for lxd, the memory allocation is <=
same with cores

Reply:
yep, I used refactofing tool (from Goland), that not correctly did its job. I'll fix it manually.